common/src/main/java/org/dash/wallet/common/WalletDataProvider.kt

```diff
@@ -51,6 +51,8 @@ interface WalletDataProvider {
 
     fun observeTransactions(vararg filters: TransactionFilter): Flow<Transaction>
 
+    fun observeTransactionsWithConfidence(vararg filters: TransactionFilter): Flow<Transaction>
+
```

Comment:
I didn't like having to add another method, but it was more simple than adding  Boolean to `observeTransactions`

Perhaps this would have worked:
`fun observeTransactions(observeTxConf: Boolean = false, vararg filters: TransactionFilter): Flow<Transaction>`

Doing this required changing every method call, mostly in the crowdnode module:
`fun observeTransactions(observeTxConf: Boolean, vararg filters: TransactionFilter): Flow<Transaction>`

I wanted to minimize the number of changes associated with turning ON the transaction confidence listener.

Reply:
This will work fine. 
We can probably also change `vararg` to a simple list later - there isn't really any need to be fancy here - and then add a default boolean argument.